dcc-import-gene/src/main/java/org/icgc/dcc/imports/gene/core/GeneConstructor.java

```diff
@@ -0,0 +1,349 @@
+        val entry = parseLine(s);
+        if (("gene".equals(asText(entry, "type")))) {
+          if (geneNode != null) {
+            finalizeAndWriteGene();
+          }
+          geneNode = constructGeneNode(entry);
+        } else if ("transcript".equals(asText(entry, "type"))) {
+          if (curTranscript != null) {
+            finalizeTranscript();
+          }
+          curTranscript = constructTranscriptNode(entry);
+        } else if ("exon".equals(asText(entry, "type"))) {
+          exons.add(constructExonNode(entry));
+        } else if ("CDS".equals(asText(entry, "type"))) {
+          ((ObjectNode) exons.get(exons.size() - 1)).put("cds", entry);
+        } else if ("start_codon".equals(asText(entry, "type"))) {
+          curTranscript.put("start_exon", exons.size() - 1);
+        } else if ("stop_codon".equals(asText(entry, "type"))) {
+          curTranscript.put("end_exon", exons.size() - 1);
+        }
+      }
+    }
+  }
+
+  /**
+   * Helper for setting the final transcript values
+   */
+  private void finalizeAndWriteGene() {
+    // Finish with the current transcript.
+    finalizeTranscript();
+    geneNode.put("transcripts", transcripts);
+    geneNode.put("external_db_ids", externalIds().get(asText(geneNode, "_gene_id")));
+    writeGene(geneNode);
+    // Reset the current state
+    curTranscript = null;
+    transcripts = DEFAULT.createArrayNode();
+  }
+
+  private void finalizeTranscript() {
+    curTranscript.put("exons", exons);
+    val trans =
+        postProcessTranscript(curTranscript, asText(geneNode, "strand"), asText(geneNode, "canonical_transcript_id"));
+    transcripts.add(trans);
+    exons = DEFAULT.createArrayNode();
+  }
+
+  private void writeGene(@NonNull ObjectNode value) {
+    writer.writeFiles(value);
+  }
+
+  /**
+   * Responsible for parsing the lines of the gtf file and producing generic ObjectNodes
+   * @param String representing the current line in the gtf file
+   * @return ObjectNode representation of the gtf row.
+   */
+  private ObjectNode parseLine(@NonNull String s) {
+    String[] line = TAB.splitToList(s).stream().toArray(String[]::new);
+    val seqname = line[0].trim();
+    val source = line[1].trim();
+    val type = line[2].trim();
+    val locStart = line[3].trim();
+    val locEnd = line[4].trim();
+
+    char strand = line[6].trim().charAt(0);
+    int locationStart = Integer.parseInt(locStart);
+    int locationEnd = Integer.parseInt(locEnd);
+    if (locationStart > locationEnd) {
+      int location = locationStart;
+      locationStart = locationEnd;
+      locationEnd = location;
+    }
+
+    val negative = locationStart <= 0 && locationEnd <= 0;
+    assert strand == '-' == negative;
+
+    val attributes = line[8];
+    val attributeMap = parseAttributes(attributes);
+
+    int strandNumber = 0;
+    if (strand == '+') {
+      strandNumber = 1;
+    } else if (strand == '-') {
+      strandNumber = -1;
+    }
```

Comment:
Replace with:

`val strandNumber = convertStand(strand)`

```java
private static int convertStand(char strand){
if (strand == '+') {
  return 1
} else if (strand == '-') {
 return -1;
} else {
 return 0;
}
```

This reduces noise in the mainline code making the overall logic more apparent (less noise).